query/view/boost_exact_matches.js

```diff
@@ -0,0 +1,48 @@
+
+var peliasQuery = require('pelias-query'),
+    searchDefaults = require('../search_defaults');
+
+/**
+  This view (unfortunately) requires autocomplete to use the phrase.* index.
+
+  ideally we wouldn't need to use this, but at time of writing we are unable
+  to distinguish between 'complete tokens' and 'grams' in the name.* index.
+
+  this view was introduced in order to score exact matches higher than partial
+  matches, without it we find results such as "Clayton Avenue" appearing first
+  in the results list for the query "Clay Av".
+
+  the view uses some of the values from the 'search_defaults.js' file to add an
+  additional 'SHOULD' condition which scores exact matches slighly higher
+  than partial matches.
+**/
+
+module.exports = function( vs ){
+
+  // make a copy of the variables so we don't interfere with the values
+  // passed to other views.
+  var vsCopy = new peliasQuery.Vars( vs.export() );
+
+  // copy phrase:* values from search defaults
+  vsCopy.var('phrase:analyzer').set(searchDefaults['phrase:analyzer']);
+  vsCopy.var('phrase:field').set(searchDefaults['phrase:field']);
+
+  // split the 'input:name' on whitespace
+  var name = vs.var('input:name').get(),
+      tokens = name.split(' ');
```

Comment:
perhaps split on any whitespace so that multiple spaces, tabs, or any other weird stuff is handled ok? Or would that be cleaned up by a sanitizer before this?